cpp/benchmarks/stream_compaction/apply_boolean_mask_benchmark.cu

```diff
@@ -0,0 +1,154 @@
+  b->Unit(benchmark::kMillisecond);
+  for (int size = tenK; size <= hundredM; size *= 10)
+    b->Args({size, fifty_percent});
+}
+
+template <typename T>
+T random_int(T min, T max)
+{
+  static unsigned seed = 13377331;
+  static std::mt19937 engine{seed};
+  static std::uniform_int_distribution<T> uniform{min, max};
+
+  return uniform(engine);
+}
+
+template <class T>
+class ApplyBooleanMask : public cudf::benchmark {
+public:
+  using TypeParam = T;
+};
+
+template <typename T>
+void calculate_bandwidth(benchmark::State& state, gdf_size_type num_columns) {
+  const gdf_size_type column_size{static_cast<gdf_size_type>(state.range(0))}; 
+  const gdf_size_type percent_true{static_cast<gdf_size_type>(state.range(1))}; 
+
+  float fraction = percent_true / 100.f;
+  gdf_size_type column_size_out = fraction * column_size;
+  int64_t mask_size = sizeof(cudf::bool8) * column_size + gdf_valid_allocation_size(column_size);
+  int64_t validity_bytes_in  = (fraction >= 1.0f/32) ? 
+    gdf_valid_allocation_size(column_size) :
+    4 * column_size_out;
+  int64_t validity_bytes_out = gdf_valid_allocation_size(column_size_out);
+  int64_t column_bytes_out = sizeof(T) * column_size_out;
+  int64_t column_bytes_in = column_bytes_out;
+
+  int64_t bytes_read = (column_bytes_in + validity_bytes_in) * num_columns + // reading columns
+    mask_size; // reading boolean mask
+  int64_t bytes_written = (column_bytes_out + validity_bytes_out) * num_columns; // writing columns
+
+  state.SetItemsProcessed(state.iterations() * column_size * num_columns);
+  state.SetBytesProcessed(static_cast<int64_t>(state.iterations()) * 
+                          bytes_read + bytes_written);
+}
+
+} // namespace anonymous
+
+template <class T>
+void BM_apply_boolean_mask(benchmark::State& state, gdf_size_type num_columns) {
+  using wrapper = cudf::test::column_wrapper<T>;
+  using mask_wrapper = cudf::test::column_wrapper<cudf::bool8>;
+
+  const gdf_size_type column_size{static_cast<gdf_size_type>(state.range(0))}; 
+  const gdf_size_type percent_true{static_cast<gdf_size_type>(state.range(1))}; 
+
+  std::vector<cudf::test::column_wrapper<T> > columns;
+
+  for (int i = 0; i < num_columns; i++) {
+    columns.emplace_back(column_size,
+      [](gdf_index_type row) { return static_cast<T>(row); },
+      [](gdf_index_type row) { return true; });
+  }
+
+  mask_wrapper mask { column_size,
+    [&](gdf_index_type row) { 
+      return cudf::bool8{random_int(0, 100) < percent_true}; 
+    },
+    [](gdf_index_type row)  { return true; }
+  };
+
+  std::vector<gdf_column*> raw_columns(num_columns, nullptr);
+  std::transform(columns.begin(), columns.end(), raw_columns.begin(),
+                 [](wrapper &c) { return c.get(); });  
+
+  cudf::table source_table{raw_columns.data(), num_columns};
+
+  for(auto _ : state){
+    cudf::table result = cudf::apply_boolean_mask(source_table, *mask.get());
+    result.destroy();
```

Comment:
I'd think the deallocation of the result should be excluded from the benchmark timing. You could exclude this with the [`cuda_event_t`](https://github.com/rapidsai/cudf/blob/branch-0.9/cpp/benchmarks/synchronization/synchronization.hpp#L67).

Reply:
Done.